plugins/linux-pipewire/formats.c

```diff
@@ -101,6 +101,16 @@ static const struct obs_pw_video_format supported_formats[] = {
 
 	},
 #endif
+	{
+		SPA_VIDEO_FORMAT_ENCODED,
+		DRM_FORMAT_INVALID,
+		GS_UNKNOWN,
+		VIDEO_FORMAT_NONE,
+		false,
+		0,
+		"Encoded",
+	},
+
```

Comment:
I tried here https://github.com/dimtpap/obs-studio/commit/910a9569eaac1c2890897982c363ead1982ba25f

The problem is that the format user facing property works with spa_video_format and the distinction between H264 and MJPEG is done with spa_media_subtype

Reply:
I have a idea for that but when I tried to improve the UX with https://github.com/tytan652/obs-studio/commit/3d15eb4b5259fe0e7aa92d09bee4beb6996a77a0  (similar to #10895) which would allow to add a line per codec in your case.

I discovered that, with the upstream code as it is, the format is never enforced and the user can end up with a format not matching the format selected.

If the camera case the format must not be negotiated and follow the user selection.
And PipeWire is not my cup of tea.